src/basic/errno-util.h

```diff
@@ -70,6 +70,13 @@ static inline int errno_or_else(int fallback) {
         return -abs(fallback);
 }
 
+/* For send()/recv() or read()/write(). */
+static inline bool ERRNO_IS_INTERRUPT(int r) {
```

Comment:
Maybe ERRNO_IS_TRANSIENT()? because these two errors are transient after all, you can call the syscall again and it will do useful stuff